helpers/swap/OneInchCallMock.ts

```diff
@@ -1,17 +1,38 @@
 import BigNumber from 'bignumber.js'
 
 import { testMarketPrice } from '../../test/config'
+import { amountFromWei, amountToWei } from '../utils'
 
 export const oneInchCallMock =
-  (marketPrice: BigNumber = new BigNumber(testMarketPrice)) =>
+  (
+    marketPrice: BigNumber = new BigNumber(testMarketPrice),
+    precision: { from: number; to: number } = { from: 18, to: 18 },
+    debug = false,
+  ) =>
   async (from: string, to: string, amount: BigNumber, slippage: BigNumber) => {
+    // EG FROM WBTC 8 to USDC 6
+    // Convert WBTC toWei at 18
+    // Apply market price
+    // Convert result back to USDC at 6
+    const precisionAdjustedToAmount = amountToWei(
```

Comment:
That looks like complication to me.

`EG: 0.01 BTC  at 16000 USD price`
`0.01*16000 = 160`
`fromToken : amountToWei(0.01,6)`
`toToken: amountToWei(160,8)`

Reply:
Updated.